### PR TITLE
[SPARK-38550][DOCS][FOLLOWUP] Improve the documentation of Diagnostic disk store

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

* Add the conf `spark.appStatusStore.diskStoreDir` in configuration.md
* This diagnostic API requires setting `spark.appStatusStore.diskStoreDir` for storing the diagnostic information.
* Change the doc of conf `spark.appStatusStore.diskStoreDir`

### Why are the changes needed?

Improve the documentation so that users can know it better

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Preview:
image

image

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @LinhongLiu @cloud-fan @dongjoon-hyun

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala [hunk 1]
```diff
@@ -73,8 +73,8 @@ private[spark] object Status {
 
   val DISK_STORE_DIR_FOR_STATUS =
     ConfigBuilder("spark.appStatusStore.diskStoreDir")
-      .doc("Local directory where to store app status. " +
-           "It's an alternative to the in-memory kv store")
+      .doc("Local directory where to store diagnostic information of SQL executions. " +
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/internal/config/Status.scala:76]: @linhongliu-db  do you plan to add put more data besides the diagnostic info into the local dir?

Comment (linhongliu-db) [reply core/src/main/scala/org/apache/spark/internal/config/Status.scala:76]: diagnostic should be enough for now.
but in other future, I'd like to see the disk store keep all app status and the in-memory store only be used as a cache layer.

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/internal/config/Status.scala:76]: > I'd like to see the disk store keep all app status and the in-memory store only be used as a cache layer.

This requires a lot of refactoring...
Let's make it specified for now. Currently, "store app status" is so general and confusing.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @gengliangwang .

File: core/src/main/scala/org/apache/spark/internal/config/Status.scala (hunk 1 above)
Comment (linhongliu-db) [reply core/src/main/scala/org/apache/spark/internal/config/Status.scala:76]: sure. make sense to me.

**Pull request closed without merging**